src/app/GroupLayer.tsx

```diff
@@ -0,0 +1,272 @@
+/** @jsxImportSource @emotion/react */
+
+// This file contains code related to the group step. Earlier, we had
+// code related to each step spread across multiple files, the reason
+// was that we architected the intial technical spike around tools.
+// Now, we have a single file for each step. Eventhough we have a lot
+// of repetitive code because of this change, eventually, we will
+// refactor and fix it.
+
+import { Dispatch } from "redux";
+import { GroupLayerActionMenu } from "../components/ActionMenu";
+import Annotation, {
+  AnnotationBeingCreated,
+  AnnotationProps,
+  CreationState,
+  TranslucentBox,
+} from "./Annotation";
+import { CreateAnnotationAttr, NO_OP, RenderAnnotationsHandler } from "./PDF";
+import {
+  useSelector,
+  useDispatch,
+  AccessibleForm,
+  AnnotationId,
+} from "./StoreProvider";
+import React from "react";
+import Xarrow from "react-xarrows";
+import { AllTokens } from "./LabelLayer";
+
+export const groupLayerHandlers = (
+  state: AccessibleForm,
+  dispatch: Dispatch,
+  createAnnotationAttr: CreateAnnotationAttr
+) => {
+  const {
+    div: container,
+    creationState,
+    newCreationBounds,
+    resetCreationState,
+    updateCreationState,
+  } = createAnnotationAttr;
+  const { tool, selectedAnnotations } = state;
+
+  switch (tool) {
+    case "CREATE": {
+      return {
+        cursor: "crosshair",
+        creationState,
+        container,
+        onClick: NO_OP,
+        onMouseDown: newCreationBounds,
+        onMouseMove: updateCreationState,
+        onMouseLeave: resetCreationState,
+        onMouseUp: () => {
+          if (!creationState) return;
+          // The following line stops dispatch of CREATE_ANNOTATION_FROM_TOKENS if no token is selected.
```

Comment:
This is indeed a bug, but instead of trying to handle the logic in the `onClick` handler, we can move the case to the reducer and change some of this code.